src/overview/how-it-works.md

```diff
@@ -1,24 +1,43 @@
 # How it works
 
-Every time you push to a live branch or activate a cluster for a branch there are two main processes that happen: *Build* and *Deploy*.  The build process looks through the configuration files in your repository and assembles the necessary LXC containers.  The deploy process makes those containers live, replacing the previous versions, with virtually no interruption in service.
+## Building and deploying applications
+
+Every time you push to a live branch (a git branch with an active environment attached to it) or activate an [environment](administration/web/environments.md) for a branch, there are two main processes that happen: *Build* and *Deploy*.  
+
+1.  The build process looks through the configuration files in your repository and assembles the necessary containers.  
+2. The deploy process makes those containers live, replacing the previous versions, with virtually no interruption in service.
 
 ![The Build and Deploy pipeline](/images/build-pipeline.svg)
 
+## Good Practices and Constraints
+
+Platform.sh promotes the practice of using a build-oriented structure; This  means that instead of committing to your Git repository the final state of your application, you would only commit the files specific to it, any frameworks, libraries and dependencies should be pulled at build-time as should any assets that should be compiled. 
```

Comment:
This is still the good practice. Commit a lock file, not the libs. Add a comment if you'd like that sometimes you don't have a choice. Sometimes you vendorize. That's ok too.